### ISSUE TITLE
Home manager permissions/structure questions

### ISSUE THREAD

**Issue by ersinakinci** [Misterio77/nix-starter-configs#65]
Hi there, thanks for putting this together! I'm returning to NixOS now that they've implemented flakes and the community has matured a bit, and I came across your templates while reading https://nixos-and-flakes.thiscute.world/nixos-with-flakes/start-using-home-manager. I'm still very new to everything.

Question about your home-manager implementation in your standard template. When I try to run `home-manager switch --flake /etc/#...` after having installed it with `nix shell nixpkgs#home-manager`, I get a permissions error: `error: opening file '/etc/.pwd.lock': Permission denied`. Am I supposed to run `sudo home-manager switch --flake /etc/#...`?

If I'm supposed to use `sudo` (btw this should be documented if it's the case), doesn't that imply that my home-manager-managed dotfiles and packages will all be owned by root? Is that intentional? Seems like it runs contrary to the spirit of home-manager, which I imagine is supposed to allow for users to manage their own environments. Maybe it would even lead to problems with certain packages that expect certain permissions on certain files.

Or is it actually a good thing? If everything is owned by root, even the dotfiles, then the user's processes can't change them ad hoc. Everything is forced to be managed through the flake. Now that I think about it, I guess that's how it works when you configure home-manager as a NixOS module...?

Maybe it's relevant, I don't know, but in the [home-manager manual](https://nix-community.github.io/home-manager/index.xhtml), it says that if you're doing a standalone installation that you should add your user to the [`nix.settings.allowed-users`](https://nixos.org/manual/nixos/stable/options.html#opt-nix.settings.allowed-users) option. Shouldn't that be part of this template? How would that interact with permissions?

So my underlying question is, what's the philosophy/rationale behind the structure you've selected for home-manager? Why not manage everything from `~/.config/home-manager/flake.nix`?

**Comment by ersinakinci** [comment]
I went ahead and tried running `sudo home-manager switch --flake /etc#...` and it resolved the permission error, but now I got a new one: `Error: USER is set to "root" but we expect "..."`

**Comment by Brisingr05** [comment]
The "NixOS and Flakes Book" is using the Home Manager NixOS module, i.e. running `sudo nixos-rebuild switch --flake` rebuilds NixOS as well as Home Manager. Misterio's templates use the standalone version of Home Manager, which requires you to run a separate command, i.e. `home-manager switch --flake` to rebuild Home Manager, which needs to be run without sudo. The standalone version's default config location is `~/.config/home-manager`, so sudo privileges aren't required.

You can move the files from under `/etc/nixos` to some location in your user's home directory. Then you will require sudo privileges only for rebuilding NixOS, not Home Manager.

**Comment by ersinakinci** [comment]
Thanks @Brisingr05. In the end, I kept the Home-Manager-as-a-module setup and I just run `sudo nixos-rebuild switch` from within `/etc/nixos`.

> Seems like it runs contrary to the spirit of home-manager, which I imagine is supposed to allow for users to manage their own environments.

I changed my mind about this. Centralizing my user's configuration makes it super easy to keep everything in one place and back up my whole setup to a remote git repo. It's more painful/annoying in terms of keystrokes (lots of `sudo`), but I think for now it's fine.

And at any rate, as I found out the hard way, Home Manager only creates symlinks to your dotfiles instead of copying them into your home folder. Your actual configs are in the Nix store, which are mounted as a read-only immutable filesystem. I think that this design choice blunts any advantage one might gain by running a standalone configuration. It doesn't matter whether the files are owned by root or the user, they're just symlinks and the target files that they point to can't be changed by either.
